globalinfotek/linux-dev-practice
# Lide 13 (Exercises - Kernel Mode - Hello World)

NOTES:

- There were some updates to the CDE in this branch so ensure you provision the CDE (e.g., `vagrant up --provision`)
- Do *not* merge this before LIDE-004 gets merged in.  I forked from LIDE-004 to do this work.

1. Read the documentation and verify it for readability, accuracy, etc

- `exercises/2-Kernel_Mode/challenge_201/README.md`
- `exercises/2-Kernel_Mode/challenge_201/201_source.c` (File/function comments)
- `exercises/2-Kernel_Mode/challenge_201/solution/README_solution.md`

2. Verify the testing steps

- Verify the `Manual Testing` steps in `exercises/2-Kernel_Mode/challenge_201/README.md`
- Verify the `Automated Testing` examples in `exercises/2-Kernel_Mode/challenge_201/README.md`

3. Verify the example

Sure, the "put your code here" source file is going to fail a few tests.  But what about my example?  Is it accurate?  Is it appropriate?  Does it pass for you?  (Not all CDEs behave the same.  There are ghosts in the host OS...)

Copy paste `exercises/2-Kernel_Mode/challenge_201/solution/201_example.c` into `exercises/2-Kernel_Mode/challenge_201/201_source.c` and run the `Automated Testing`.  It *should* pass...
